tests/Resources/ti.ui.alertdialog.test.js

```diff
@@ -0,0 +1,104 @@
+		should(bar.getTitle()).eql('this is some text');
+		bar.title = 'other text';
+		should(bar.title).eql('other text');
+		should(bar.getTitle()).eql('other text');
+	});
+
+	// FIXME titleid doesn't seem to set title on iOS?
+	(utilities.isIOS() ? it.skip : it)('titleid', function () {
+		var bar = Ti.UI.createAlertDialog({
+			titleid: 'this_is_my_key'
+		});
+		should(bar.titleid).be.a.String;
+		should(bar.getTitleid).be.a.Function;
+		should(bar.titleid).eql('this_is_my_key');
+		should(bar.getTitleid()).eql('this_is_my_key');
+		should(bar.title).eql('this is my value'); // fails on iOS, gives undefined
+		bar.titleid = 'other text';
+		should(bar.titleid).eql('other text');
+		should(bar.getTitleid()).eql('other text');
+		should(bar.title).eql('this is my value'); // retains old value if key not found: https://jira.appcelerator.org/browse/TIMOB-23498
+	});
+
+	it('message', function () {
+		var bar = Ti.UI.createAlertDialog({
+			message: 'this is some text'
+		});
+		should(bar.message).be.a.String;
+		should(bar.getMessage).be.a.Function;
+		should(bar.message).eql('this is some text');
+		should(bar.getMessage()).eql('this is some text');
+		bar.message = 'other text';
+		should(bar.message).eql('other text');
+		should(bar.getMessage()).eql('other text');
+	});
+
+	// FIXME Get working on iOS - defaults to undefined, should be ['OK']
+	// FIXME Get working on Android - defaults to undefined, should be ['OK']
+	((utilities.isIOS() || utilities.isAndroid()) ? it.skip : it)('buttonNames', function () {
+		var bar = Ti.UI.createAlertDialog({});
+		should(bar.buttonNames).be.an.Array; // undefined on iOS and Android
+		should(bar.getButtonNames).be.a.Function;
+		should(bar.buttonNames).be.empty;
+		should(bar.getButtonNames()).be.empty;
+		bar.buttonNames = ['this','other'];
+		should(bar.buttonNames.length).eql(2);
+		should(bar.getButtonNames().length).eql(2);
+	});
+
+	// FIXME Get working on iOS - defaults to undefined, should be -1
+	// FIXME Get working on Android - defaults to undefined, should be -1
+	((utilities.isIOS() || utilities.isAndroid()) ? it.skip : it)('cancel', function (finish) {
+		var bar = Ti.UI.createAlertDialog({});
+		should(bar.cancel).be.a.Number; // undefined on iOS and Android
+		should(bar.getCancel).be.a.Function;
+		bar.cancel = 1;
+		should(bar.cancel).eql(1);
+		should(bar.getCancel()).eql(1);
+	});
+	
+// Skip on Android, since it's an iOS-only property
+(utilities.isAndroid() ? it.skip : it)('tintColor', function () {
+	var bar = Ti.UI.createAlertDialog({
+		tintColor: 'red'
+	});
+	
+	// Check getter
+	should(bar.tintColor).be.a.String;
+	should(bar.getTintColor).be.a.Function;
+	should(bar.tintColor).eql('red');
+	should(bar.getTintColor()).eql('red');
+	
+	// Set new value
+	bar.tintColor = '#f00';
+	
+	// Check getter again
+	should(bar.tintColor).be.a.String;
+	should(bar.getTintColor).be.a.Function;
+	should(bar.tintColor).eql('#f00');
+	should(bar.getTintColor()).eql('#f00');
```

Comment:
Is there a setter method we should check exists, and can be called?

It sounds simple/dumb, but I swear on Android I saw a case where the property setter and the actual getter method were "disconnected" where you could set different values for each and they didn't update the same underlying property.